### ISSUE TITLE
x/website: elevated rate of 502s

### ISSUE THREAD

**Issue by dmitshur** [golang/go#30619]
I am seeing a relatively high frequently of 502s on the golang.org website:

```
$ time curl 'https://golang.org/_ah/health'
ok
real	0m0.209s
user	0m0.025s
sys	0m0.010s
$ time curl 'https://golang.org/_ah/health'
ok
real	0m0.142s
user	0m0.025s
sys	0m0.007s
$ time curl 'https://golang.org/_ah/health'
ok
real	0m0.141s
user	0m0.025s
sys	0m0.007s
$ time curl 'https://golang.org/_ah/health'

<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>

real	0m9.192s
user	0m0.017s
sys	0m0.006s
$ time curl 'https://golang.org/_ah/health'
ok
real	0m0.438s
user	0m0.022s
sys	0m0.007s
$ time curl 'https://golang.org/_ah/health'
ok
real	0m0.142s
user	0m0.024s
sys	0m0.007s
```

I suspect it's what's causing https://github.com/golang/lint/issues/440. /cc @agnivade

It seems to be happening on previously deployed versions of golang.org, not just the current one.

The last error that https://tip.golang.org/_tipstatus ran into is also 502 related, but when cloning from go.googlesource.com/go:

```
error=builder.Init: checkout of go: fetch: fatal: unable to access 'https://go.googlesource.com/go/': The requested URL returned error: 502
```

Maybe it's related to some higher-level component and not the website specifically? I don't have enough information to tell yet. /cc @bradfitz @andybons

**Comment by agnivade** [comment]
Thanks for the quick turnaround @dmitshur !

**Comment by dmitshur** [comment]
I can reliably get frequent 502s from golang.org and its previously deployed versions right now, but not other sites like tour.golang.org, so it seems contained to the main website from what I can tell so far.

I remember last time something like this happened it was because the website was misconfigured regarding its use of index, and ended up pegging the CPU at 100%, etc. That's just for reference; I don't know what's happening now, so this needs more investigation.

**Comment by dmitshur** [comment]
From the GCP console, I see there's now about one 5xx code out of every 50 responses. The elevated 5xx rates began at roughly 9:06 PM eastern and have stayed consistently there.

**Comment by dmitshur** [comment]
I've restarted the instance and it seems better now; I can't reproduce the 502s anymore.

It's still not as smooth as pre-9:06 pm; some occasional requests to `https://golang.org/_ah/health` take 4.8~ seconds instead of the usual 0.150~ seconds. So this needs more investigation, but at least https://github.com/golang/lint/issues/440 shouldn't be happening now.

**Comment by dmitshur** [comment]
The immediate issue is resolved, so I'll remove Soon label.

**Comment by dmitshur** [comment]
We've done more investigation here, and it turned out the root cause was external to the golang.org server. It was a temporary issue affecting another networking component that has since been resolved.

I've watched the golang.org server, and other than the elevated 502 rate during the affected period on Tuesday night (9:06 pm to 12:36 am), the issue has no re-occurred:

![image](https://user-images.githubusercontent.com/1924134/54061971-5e71d700-41d1-11e9-9ebf-25124fba11bf.png)

Closing, since there's nothing more to do here. Huge thanks to @broady for helping investigate and finding the root cause!
